quasar-core/src/main/java/co/paralleluniverse/fibers/suspend/Stacks.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Quasar: lightweight threads and actors for the JVM.
+ * Copyright (c) 2021, R3 Ltd. All rights reserved.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation
+ *
+ *   or (per the licensee's choosing)
+ *
+ * under the terms of the GNU Lesser General Public License version 3.0
+ * as published by the Free Software Foundation.
+ */
+package co.paralleluniverse.fibers.suspend;
+
+import java.util.function.Supplier;
+
+@SuppressWarnings("unused")
+public final class Stacks {
+    private static volatile Supplier<? extends StackOps> GET_STACK;
+
+    public static StackOps getStack() {
+        final Supplier<? extends StackOps> getter = GET_STACK;
+        return getter != null ? getter.get() : null;
```

Comment:
I wonder to what extent it makes sense to  `throw RuntimeException("Quasar not correctly initialised")` rather than returning null ?  Does it actually surface I wonder or lost in the wilderness of a weird Quasar NPEs etc?

Reply:
The underlying `Stack.getStack()` function can return `null`:
```java
public static Stack getStack() {
    final Fiber<?> currentFiber = Fiber.currentFiber();
    return currentFiber != null ? currentFiber.stack : null;
}
```
and so I don't believe that `Stacks.getStack()` also being able to return `null` should be a problem.

Reply:
Well, we have seen really annoying NPEs when quasar gets a bit confused and wondered if it was an opportunity to tidy that up a bit, at least where the initialisation hasn't worked.

Reply:
The instrumentation code does check whether `getStack()` returns `null`. Did you trace your NPEs back to `Stack` operations?

There _will_ now be a delay before the OSGi framework can load the `quasar-core-osgi` bundle, during which time the agent will still be instrumenting classes. I think that handling this period gracefully would be a Good Thing :tm:.

Reply:
Yes, I'd not thought about the fact that instrumented code is executed (most likely) on Threads before any Fibers exist.  Good point and my suggestion would probably break Corda.  Will mark as resolved!